test/test-helper.js

```diff
@@ -817,19 +818,33 @@ Ccm.prototype.waitForUp = function (callback) {
   var retryCount = 0;
   var self = this;
   utils.whilst(function () {
-    return !started && retryCount < 10;
+    return !started && retryCount < 60;
   }, function iterator (next) {
-    self.exec(['node1', 'showlog'], function (err, info) {
-      if (err) {
-        return next(err);
-      }
-      var regex = /Starting listening for CQL clients/mi;
-      started = regex.test(info.stdout.join(''));
+    var nodesTested = 0;
+    var ping = true;
+    utils.whilst(function () {
+      return nodesTested < self.nodeLength;
+    }, function pingNodes (nextIp) {
+      var pingIp = '127.0.0.' + (nodesTested + 1);
+      helper.trace('Attempt to ping ' + pingIp);
+      nodesTested++;
+      var conn = net.createConnection(9042, pingIp)
+        .on('error', function() {
+          ping = false;
+          nextIp();
+        })
+        .on('connect', function() {
+          ping = ping && true;
+          conn.end();
```

Comment:
On a established connection, if the client ends (FIN) the connection before sending the STARTUP message, C*will error out before closing it.
This results in `'connect'` event, followed by `'error'`.

The fix should be something like this:
```
con
  .on('error', function () {
    if (ping) {
      // The connection was established, don't mind about errors after `socket.end()`
      return;
    }
    nextIp();
  })
  .on('connect', function () {
    ping = true;
    nextIp();
  });
```